packages/components/src/Actions/ActionButton/ActionButton.component.js

```diff
@@ -89,23 +93,25 @@ function ActionButton(props) {
 
 	const buttonProps = getPropsFrom(Button, rest);
 	const style = link ? 'link' : bsStyle;
-	const rClick =
-		onClick &&
-		(event =>
-			onClick(event, {
-				action: { label, ...rest },
-				model,
-			}));
-
-	const rMouseDown = event =>
-		onMouseDown(event, {
-			action: { label, ...rest },
-			model,
-		});
+	const rClick = !overlayComponent
+		? event =>
+				onClick(event, {
+					action: { label, ...rest },
+					model,
+				})
+		: null;
+
+	const rMouseDown = !overlayComponent
+		? event =>
+				onMouseDown(event, {
+					action: { label, ...rest },
+					model,
+				})
+		: null;
```

Comment:
Those blocs are hard to read.
Can you reshape this code, perhaps with something like
```
const buttonProps = getPropsFrom(Button, rest);
const customProps = { style: (link ? 'link' : bsStyle), ... };

if (!overlayComponent) {
    customProps.onClick = ...
    customProps.onMouseDown = ...
}
```